tools/distpartitioning/data_shuffle.py

```diff
@@ -643,82 +757,102 @@ def gen_dist_partitions(rank, world_size, params):
     memory_snapshot("ReorderNodeFeaturesComplete: ", rank)
 
     #sort edge_data by etype
-    sorted_idx = edge_data[constants.ETYPE_ID].argsort()
-    for k, v in edge_data.items():
-        edge_data[k] = v[sorted_idx]
-    sorted_idx = None
+    for local_part_id in range(params.num_parts//world_size):
+        sorted_idx = edge_data[constants.ETYPE_ID+"/"+str(local_part_id)].argsort()
+        for k, v in edge_data.items():
+            if k.endswith(str(local_part_id)):
+                edge_data[k] = v[sorted_idx]
+        sorted_idx = None
     gc.collect()
 
-    shuffle_global_eid_start = assign_shuffle_global_nids_edges(rank, world_size, edge_data)
+    shuffle_global_eid_offsets = assign_shuffle_global_nids_edges(rank, world_size, params.num_parts, edge_data)
     logging.info(f'[Rank: {rank}] Done assigning global_ids to edges ...')
     memory_snapshot("ShuffleGlobalID_Edges_Complete: ", rank)
 
     #Shuffle edge features according to the edge order on each rank.
     for etype_name in etypes:
         featnames = get_etype_featnames(etype_name, schema_map)
         for featname in featnames:
-            feature_key = etype_name+'/'+featname
-            assert feature_key in rcvd_global_eids
-            global_eids = rcvd_global_eids[feature_key]
+            for local_part_id in range(params.num_parts//world_size):
+                feature_key = etype_name+'/'+featname+"/"+str(local_part_id)
+                assert feature_key in rcvd_global_eids
+                global_eids = rcvd_global_eids[feature_key]
 
-            _, idx1, _ = np.intersect1d(edge_data[constants.GLOBAL_EID], global_eids, return_indices=True)
-            shuffle_global_ids = edge_data[constants.SHUFFLE_GLOBAL_EID][idx1]
-            feature_idx = shuffle_global_ids.argsort()
+                _, idx1, _ = np.intersect1d(edge_data[constants.GLOBAL_EID+"/"+str(local_part_id)], global_eids, return_indices=True)
+                shuffle_global_ids = edge_data[constants.SHUFFLE_GLOBAL_EID+"/"+str(local_part_id)][idx1]
+                feature_idx = shuffle_global_ids.argsort()
 
-            rcvd_edge_features[feature_key] = rcvd_edge_features[feature_key][feature_idx]
-
-    for k, v in rcvd_edge_features.items():
-        logging.info(f'[Rank: {rank}] key: {k} v: {v.shape}')
+                rcvd_edge_features[feature_key] = rcvd_edge_features[feature_key][feature_idx]
 
     #determine global-ids for edge end-points
-    edge_data = lookup_shuffle_global_nids_edges(rank, world_size, edge_data, id_lookup, node_data)
+    edge_data = lookup_shuffle_global_nids_edges(rank, world_size, params.num_parts, edge_data, id_lookup, node_data)
     logging.info(f'[Rank: {rank}] Done resolving orig_node_id for local node_ids...')
     memory_snapshot("ShuffleGlobalID_Lookup_Complete: ", rank)
 
+    def prepare_local_data(src_data, local_part_id):
+        local_data = {}
+        for k, v in src_data.items():
+            if k.endswith(str(local_part_id)):
```

Comment:
same issue with previous comment

Reply:
made the same change here as described in the last comment.